sound/soc/sof/amd/acp-loader.c

```diff
@@ -215,7 +216,7 @@ int acp_sof_dsp_run(struct snd_sof_dev *sdev)
 	dev_dbg(sdev->dev, "ACP_DSP0_RUNSTALL : 0x%0x\n", val);
 
 	/* Some platforms won't support fusion DSP,keep offset zero for no support */
-	if (desc->fusion_dsp_offset) {
+	if (desc->fusion_dsp_offset && adata->enable_fw_debug) {
```

Comment:
is this correct @vijendarmukunda previously you write to the register even when the module param wasnt set but now this is dependent on the param. Did you mean || here?

Reply:
@ranj063 : fusion_dsp_offset register exists from ACP 6.0 versions. This register needs to be programmed to launch the FW debug session. As this register exists from ACP 6.0 version, we have added a feild for this register in descriptor. 
For older platforms, as this register doesn't exist, desc->fusion_dsp_offset value will be zero. Instead of all time, enabling the fusion_dsp_offset register, we want to provide provision, when FW debug is required then only this register needs to be programmed. Thats why we have added extra check enable_fw_debug.

Reply:
Thanks for the explanation @vijendarmukunda